src/Spout/Writer/WriterAbstract.php

```diff
@@ -305,35 +344,32 @@ public function addRowsWithStyle(array $dataRows, $style)
             throw new InvalidArgumentException('The "$style" argument must be a Style instance and cannot be NULL.');
         }
 
-        $this->setRowStyle($style);
-        $this->addRows($dataRows);
-        $this->resetRowStyleToDefault();
+        $this->addRows(array_map(function ($row) use ($style) {
```

Comment:
We can probably do better than that, as the row is traversed twice here. Maybe just call `addRow` in a foreach loop? as style is not handled the same way, that may work